Move to go-jsonnet and lock tooling

dgrisonnet opened the pull request:
Fix #479 

Step taken:
- ~~move to go 1.14 (if necessary)~~
- lock tooling versions in go.mod
- use go-jsonnet in version from commit 6f135f75bc0e or newer workig one (at the time of writing, there is no release with jsonnetfmt included)
- add make targets to build tools in a project path instead of system path
- remove generate-in-docker target
- **remove test-in-docker target**

cc @paulfantom

lilic commented:
Are there any downsides to us merging this now, in terms of CMO?

lilic commented on .travis.yml:
```diff
@@ -11,14 +11,11 @@ cache:
-    script: make --always-make generate-in-docker && git diff --exit-code
+    script: make --always-make generate && git diff --exit-code
   - name: Run tests
-    script: make --always-make test-in-docker
```
This needs to be updated https://github.com/coreos/kube-prometheus#contributing but I am sure there are a couple of other places.

dgrisonnet replied:
Good catch, I forgot to remove the occurrence of `test-in-docker` and `generate-in-docker` in the documentation

dgrisonnet replied:
It was actually the only place.

paulfantom commented:
> Are there any downsides to us merging this now, in terms of CMO?

No, there aren't any I can think of.

paulfantom commented on Makefile:
```diff
@@ -1,66 +1,53 @@
-	go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb@v0.3.1
+$(TOOLING): $(BIN_DIR)
+	@echo Installing tools from scripts/tools.go
+	@cat scripts/tools.go | grep _ | awk -F'"' '{print $$2}' | GOBIN=$(BIN_DIR) xargs -tI % go install %
```
Won't `go install` put golang binaries in `$GOPATH/bin` and override other system binaries?

dgrisonnet replied:
The reason behind this change is that go build -o as a different behavior in 1.12. From what I understood, in 1.12 you need to specify the name of the file at the destination _$(BIN_DIR)/jb_. Since it was not convenient I chose to use go install instead.

dgrisonnet commented:
I updated a few things in order to make the CI pass with Go 1.12:

- use `go install` instead of `go build` to build the tools
- export `GO111MODULE=on` in the Makefile
- updated PATH in `test.sh` in order to use project tooling

PTAL @paulfantom

lilic commented:
> I updated a few things in order to make the CI pass with Go 1.12:

Why not bump to 1.14 or at least 1.13?

lilic commented:
Thanks!

lilic commented:
> We can move to go 1.13, but I wouldn't consider it a blocker for this PR

Lets open a follow to move to 1.14 then :) 🎉
